moto/secretsmanager/models.py

```diff
@@ -413,24 +419,44 @@ def rotate_secret(
         old_secret_version = secret.versions[secret.default_version_id]
         new_version_id = client_request_token or str(uuid.uuid4())
 
+        # We add the new secret version as "pending". The previous version remains
+        # as "current" for now. Once we've passed the new secret through the lambda
+        # rotation function (if provided) we can then update the status to "current".
         self._add_secret(
             secret_id,
             old_secret_version["secret_string"],
             description=secret.description,
             tags=secret.tags,
             version_id=new_version_id,
-            version_stages=["AWSCURRENT"],
+            version_stages=["AWSPENDING"],
         )
-
         secret.rotation_lambda_arn = rotation_lambda_arn or ""
         if rotation_rules:
             secret.auto_rotate_after_days = rotation_rules.get(rotation_days, 0)
         if secret.auto_rotate_after_days > 0:
             secret.rotation_enabled = True
 
-        if "AWSCURRENT" in old_secret_version["version_stages"]:
-            old_secret_version["version_stages"].remove("AWSCURRENT")
+        # Begin the rotation process for the given secret by invoking the lambda function.
+        if secret.rotation_lambda_arn:
+            request_headers = {}
```

Comment:
The lambda-import should be done as late as possible, so inside this if-statement.
That way, other people can still call other SecretsManager-methods without needing the AWSLambda dependencies